### PR TITLE
[improve][broker] Make some operation deduplication methods in Namespaces async.

### PR DESCRIPTION
**Pull request by shibd**
### Motivation
See PIP https://github.com/apache/pulsar/issues/14365 and change tracker https://github.com/apache/pulsar/issues/15043.

Make NamespacesBase `getDeduplication / modifyDeduplication / removeDeduplication` methods to pure async.

### Documentation
- [x]  no-need-doc

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java [hunk 1]
```diff
@@ -513,10 +513,17 @@ public void removeNamespaceAntiAffinityGroup(@PathParam("property") String prope
...
-            @PathParam("namespace") String namespace, boolean enableDeduplication) {
+    public void modifyDeduplication(@Suspended AsyncResponse asyncResponse, @PathParam("property") String property,
+                                    @PathParam("cluster") String cluster, @PathParam("namespace") String namespace,
+                                    boolean enableDeduplication) {
         validateNamespaceName(property, cluster, namespace);
-        internalModifyDeduplication(enableDeduplication);
+        internalModifyDeduplicationAsync(enableDeduplication)
+                .thenAccept(__ -> asyncResponse.resume(Response.ok().build()))
+                .exceptionally(ex -> {
+                    log.error("Fail modify broker deduplication config for namespace {}", namespaceName, ex);
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java:523]:
```suggestion
                    log.error("Failed to modify broker deduplication config for namespace {}", namespaceName, ex);
```

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java [hunk 2]
```diff
@@ -411,32 +411,53 @@ public void removeSubscriptionExpirationTime(@PathParam("tenant") String tenant,
...
             @ApiResponse(code = 404, message = "Tenant or cluster or namespace doesn't exist") })
-    public Boolean getDeduplication(@PathParam("tenant") String tenant, @PathParam("namespace") String namespace) {
+    public void getDeduplication(@Suspended AsyncResponse asyncResponse, @PathParam("tenant") String tenant,
+                                 @PathParam("namespace") String namespace) {
         validateNamespaceName(tenant, namespace);
-        return internalGetDeduplication();
+        internalGetDeduplicationAsync()
+                .thenAccept(deduplication -> asyncResponse.resume(deduplication))
+                .exceptionally(ex -> {
+                    log.error("Fail get broker deduplication config for namespace {}", namespace, ex);
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java:420]:
```suggestion
                    log.error("Failed to get broker deduplication config for namespace {}", namespace, ex);
```

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java [hunk 3]
```diff
@@ -411,32 +411,53 @@ public void removeSubscriptionExpirationTime(@PathParam("tenant") String tenant,
...
+                                    @PathParam("namespace") String namespace,
                                     @ApiParam(value = "Flag for disabling or enabling broker side deduplication "
                                             + "for all topics in the specified namespace", required = true)
                                             boolean enableDeduplication) {
         validateNamespaceName(tenant, namespace);
-        internalModifyDeduplication(enableDeduplication);
+        internalModifyDeduplicationAsync(enableDeduplication)
+                .thenAccept(__ -> asyncResponse.resume(Response.ok().build()))
+                .exceptionally(ex -> {
+                    log.error("Fail modify broker deduplication config for namespace {}", namespaceName, ex);
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java:440]:
```suggestion
                    log.error("Failed to modify broker deduplication config for namespace {}", namespaceName, ex);
```

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java [hunk 4]
```diff
@@ -411,32 +411,53 @@ public void removeSubscriptionExpirationTime(@PathParam("tenant") String tenant,
...
             @ApiResponse(code = 404, message = "Tenant or cluster or namespace doesn't exist") })
-    public void removeDeduplication(@PathParam("tenant") String tenant, @PathParam("namespace") String namespace) {
+    public void removeDeduplication(@Suspended AsyncResponse asyncResponse, @PathParam("tenant") String tenant,
+                                    @PathParam("namespace") String namespace) {
         validateNamespaceName(tenant, namespace);
-        internalModifyDeduplication(null);
+        internalModifyDeduplicationAsync(null)
+                .thenAccept(__ -> asyncResponse.resume(Response.ok().build()))
+                .exceptionally(ex -> {
+                    log.error("Fail remove broker deduplication config for namespace {}", namespaceName, ex);
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java:457]:
```suggestion
                    log.error("Failed to remove broker deduplication config for namespace {}", namespaceName, ex);
```

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java [hunk 5]
```diff
@@ -513,10 +513,17 @@ public void removeNamespaceAntiAffinityGroup(@PathParam("property") String prope
...
             @ApiResponse(code = 404, message = "Property or cluster or namespace doesn't exist") })
-    public void modifyDeduplication(@PathParam("property") String property, @PathParam("cluster") String cluster,
-            @PathParam("namespace") String namespace, boolean enableDeduplication) {
+    public void modifyDeduplication(@Suspended AsyncResponse asyncResponse, @PathParam("property") String property,
+                                    @PathParam("cluster") String cluster, @PathParam("namespace") String namespace,
+                                    boolean enableDeduplication) {
         validateNamespaceName(property, cluster, namespace);
-        internalModifyDeduplication(enableDeduplication);
+        internalModifyDeduplicationAsync(enableDeduplication)
+                .thenAccept(__ -> asyncResponse.resume(Response.ok().build()))
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java:521]:
```suggestion
                .thenAccept(__ -> asyncResponse.resume(Response.noContent().build()))
```

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java [hunk 6]
```diff
@@ -411,32 +411,53 @@ public void removeSubscriptionExpirationTime(@PathParam("tenant") String tenant,
...
-    public void modifyDeduplication(@PathParam("tenant") String tenant, @PathParam("namespace") String namespace,
+    public void modifyDeduplication(@Suspended AsyncResponse asyncResponse, @PathParam("tenant") String tenant,
+                                    @PathParam("namespace") String namespace,
                                     @ApiParam(value = "Flag for disabling or enabling broker side deduplication "
                                             + "for all topics in the specified namespace", required = true)
                                             boolean enableDeduplication) {
         validateNamespaceName(tenant, namespace);
-        internalModifyDeduplication(enableDeduplication);
+        internalModifyDeduplicationAsync(enableDeduplication)
+                .thenAccept(__ -> asyncResponse.resume(Response.ok().build()))
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java:438]:
```suggestion
                .thenAccept(__ -> asyncResponse.resume(Response.noContent().build()))
```

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java [hunk 7]
```diff
@@ -411,32 +411,53 @@ public void removeSubscriptionExpirationTime(@PathParam("tenant") String tenant,
...
     @ApiOperation(value = "Remove broker side deduplication for all topics in a namespace")
     @ApiResponses(value = { @ApiResponse(code = 403, message = "Don't have admin permission"),
             @ApiResponse(code = 404, message = "Tenant or cluster or namespace doesn't exist") })
-    public void removeDeduplication(@PathParam("tenant") String tenant, @PathParam("namespace") String namespace) {
+    public void removeDeduplication(@Suspended AsyncResponse asyncResponse, @PathParam("tenant") String tenant,
+                                    @PathParam("namespace") String namespace) {
         validateNamespaceName(tenant, namespace);
-        internalModifyDeduplication(null);
+        internalModifyDeduplicationAsync(null)
+                .thenAccept(__ -> asyncResponse.resume(Response.ok().build()))
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java:455]:
```suggestion
                .thenAccept(__ -> asyncResponse.resume(Response.noContent().build()))
```

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java (hunk 5 above)
Comment (codelipenghui) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java:521]: From the logs, we can see that before this patch, the response HTTP code is 204.

```
2022-05-16T08:47:05,257 - INFO  - [pulsar-web-30-16:Slf4jRequestLogWriter@62] - 127.0.0.1 - - [16/May/2022:08:47:05 +0800] "POST /admin/v2/namespaces/my-property/my-ns/deduplication HTTP/1.1" 204 0 "-" "Pulsar-Java-v2.11.0-SNAPSHOT" 6
```

**Review by codelipenghui — CHANGES_REQUESTED** [review]

**Comment by Anonymitaet** [comment]
@shibd please do not delete the backticks for the doc label, or else Bot cannot recognize it and label your PR w/ `doc-info-missing`.

**Comment by shibd** [comment]
> @shibd please do not delete the backticks for the doc label, or else Bot cannot recognize it and label your PR w/ doc-info-missing.
Thanks, get.

**Comment by shibd** [comment]
@codelipenghui All suggested fixed. PTAL.

**Review by codelipenghui — APPROVED** [review]

**Review by Technoboy- — APPROVED** [review]

**Pull request merged**